ggml-opencl.c

```diff
@@ -214,12 +216,12 @@ void ggml_cl_init(void) {
         }
     }
 
-    device = NULL;
     char * GGML_OPENCL_DEVICE = getenv("GGML_OPENCL_DEVICE");
     if (GGML_OPENCL_DEVICE != NULL) {
         cl_device_id devices[NDEV];
         cl_uint num_devices;
         clGetDeviceIDs(platform, CL_DEVICE_TYPE_ALL, NDEV, devices, &num_devices);
+        CL_CHECK(err, "clGetDeviceIDs");
```

Comment:
You didn't assign to `err` here. Cleanest for functions that return the error is wrapping them in `CL_CHECK`, in my opinion.